ObjectCreation/BSpline3D.FCMacro

```diff
@@ -0,0 +1,123 @@
+#*   You should have received a copy of the GNU Lesser General Public      *
+#*   License along with FreeCAD. If not, see                               *
+#*   <https://www.gnu.org/licenses/>.                                      *
+#*                                                                         *
+#**************************************************************************/
+
+_Name__ = 'BSpline3D'
+__Comment__ = 'Create a parametric BSpline from Vertices selected in 3D view'
+__Author__ = 'CorvusCorax'
+__Version__ = '0.0.1'
+__Date__ = '2024-11-11'
+__License__ = 'LGPL-2.1-or-later'
+__Web__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D'
+__Wiki__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D/wiki'
+__Help__ = 'Select two or more vertices in the 3D view, then execute the macro.'
+__Status__ = 'beta'
+__Requires__ = 'FreeCAD >= v0.20'
+__Contact__ = 'https://github.com/CorvusCorax'
+__Communication__ = 'https://github.com/CorvusCorax/FreeCAD-BSpline3D/issues'
+__Files__ = ''
+__Xpm__ = """
+/* XPM */
+static char * XFACE[] = {
+"32 32 2 1",
+"  c #FFFFFF",
+". c #000000",
+"                                ",
+"                                ",
+"                                ",
+"                                ",
+"                                ",
+"              ........         .",
+"     ....   ...      ...    . . ",
+"    ..... ...          .   ...  ",
+"    .  ....            ..  .... ",
+"    .  ...              .  ...  ",
+"    .....               . .     ",
+"       ..                       ",
+"       .                        ",
+"      ..             ....       ",
+"      .             .....       ",
+"      ..            .  ..       ",
+"       .            .  ..       ",
+"       ..           ....        ",
+"        .                       ",
+"        ..                      ",
+"         ..                     ",
+"          ..                    ",
+"           ..                   ",
+"      .......                   ",
+"     ........                   ",
+"     .    ...                   ",
+"     .    ...                   ",
+"     .    ..                    ",
+"     ......                     ",
+"                                ",
+"                                ",
+"                                "};
+"""
+
+import FreeCAD, FreeCADGui
+import Draft
+
+def BSpline3D():
+    selection = FreeCADGui.Selection.getCompleteSelection(Gui.Selection.ResolveMode.NoResolve)
+    vertices=[]
+    for obj in selection:
+        for i,feature in enumerate(obj.SubObjects):
+            if (feature.ShapeType!="Vertex"):
+                FreeCAD.Console.PrintWarning("Selected object %s is not a Vertex (%s), using CenterOfGravity.\n"%(obj.Object.Label+"."+ obj.SubElementNames[i], feature.ShapeType)) # dirty little secret: we use CenterOfGravity for vertices as well.
+            subelements=obj.SubElementNames[i].split('.')
+            if len(subelements)==1:
```

Comment:
Maybe you could use the code from the [GetGlobalPlacement macro](https://github.com/FreeCAD/FreeCAD-macros/blob/master/Information/GetGlobalPlacement.FCMacro)? Please report if you find a case where the macro doesn't work.

Reply:
I'll have a look. I think some of the code could be reused, but it would still require code-duplication. I don't think macros can depend on other macros, can they?